### PR TITLE
Fix error too deep recursion

### PR DESCRIPTION
**Pull request by timgl**
## Changes

Fix [this sentry error](https://sentry.io/organizations/posthog2/issues/2714891907/?project=1899813&query=is%3Aunresolved+toodeeprecursion&statsPeriod=14d) when there's too many repeated team_id clauses (I think a clickhouse issues).

I _think_ it's safe to remove as none of the other filters in `parse_prop_clauses` filter on team either (like person, cohort etc).

Wasn't able to write a unit test where this failed locally.

## How did you test this code?

Existing unit tests should uncover any issues

### CODE REVIEW

**Review by EDsCODE — COMMENTED** [review]
Hmm I was going to say a good check on this would be to remove team_id as a parameter from all the parse_prop_clauses calls. It seems like there's a faulty reliance on team_id at line 57 on this function because the parameter can be passed in as optional, but the cohort getter relies on team_id to exist. Might be worth cleaning this up to be able to remove team_id from the calls

**Comment by timgl** [comment]
> Hmm I was going to say a good check on this would be to remove team_id as a parameter from all the parse_prop_clauses calls. It seems like there's a faulty reliance on team_id at line 57 on this function because the parameter can be passed in as optional, but the cohort getter relies on team_id to exist. Might be worth cleaning this up to be able to remove team_id from the calls

Done. I guess you could now pass through a cohort id of a different team, but it would never work

File: ee/clickhouse/models/property.py [hunk 1]
```diff
@@ -102,7 +99,7 @@ def parse_prop_clauses(
                 allow_denormalized_props=allow_denormalized_props,
             )
 
-            final.append(f"{filter_query} AND {table_name}team_id = %(team_id)s" if team_id else filter_query)
+            final.append(f" {filter_query}")
```
Comment (macobo) [inline ee/clickhouse/models/property.py:102]: Nit: The space isn't needed based off of my reading of this.

**Review by macobo — APPROVED** [review]
:heart: 

Longer-term parse_prop_clauses won't do any cohort lookups anyways as well. :)

File: ee/clickhouse/models/property.py (hunk 1 above)
Comment (timgl) [reply ee/clickhouse/models/property.py:102]: Hm I would agree with you but some tests did fail when I removed it.

**Pull request merged**